### ISSUE TITLE
Add testing to `commands.py`/`NodeUpdaterThread` level

### ISSUE THREAD

**Issue by ijrsvt** [ray-project/ray#10846]
### What is the problem?
There is testing for `command_runners` and for the `StandardAutoscaler` and for the cluster launcher CLI (primarily in the form of user output validation). **We should have explicit integration tests from commands.py or CLI for interactions with the node provider**

*Ray version and other system information (Python version, TensorFlow version, OS):*

### Reproduction (REQUIRED)
Please provide a script that can be run to reproduce the issue. The script should have **no external library dependencies** (i.e., use fake or mock data / environments):

If we cannot run your script, we cannot fix your issue.

- [ ] I have verified my script runs in a clean environment and reproduces the issue.
- [ ] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/latest/installation.html).

**Comment by richardliaw** [comment]
Isn't there `test_cli.py`? :)

**Comment by richardliaw** [comment]
It hits boto mocks and goes through commands.py.

**Comment by ijrsvt** [comment]
I mean we could use `test_cli`, but I would rather keep more "what the user sees" CLI tests separate from testing state changes enacted by the autoscaler (how does it manipulate the cluster).

**Comment by richardliaw** [comment]
Right - could you adjust the issue description then? "testing cluster launching from the CLI" is already satisfied by existing tests; maybe change it to "We should have explicit integration tests from commands.py or CLI for interactions with the node provider"

**Comment by ijrsvt** [comment]
@richardliaw  that sounds _great!!!_, Thanks for the help 💯!!

**Comment by ericl** [comment]
Yep, note that you can follow the pattern for testGetOrCreateMultiNodeType https://github.com/ray-project/ray/blob/master/python/ray/tests/test_resource_demand_scheduler.py#L238

(this is the *one* unit test that tests commands.py)

**Comment by ericl** [comment]
Related to https://github.com/ray-project/ray/issues/10903

**Comment by cszhu** [comment]
Hello! This P3 issue has seen no activity in at least a year. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
